### PR TITLE
Remove new preview API versions

### PR DESCRIPTION
**Pull request by danielrbradley**
Hopefully fixes https://github.com/pulumi/pulumi-azure-native/issues/2294

- Add mechanism for removing specific API verisons
- Curate preview versions added since the last provider release - so this shouldn't be a breaking change.

### CODE REVIEW

**Comment by danielrbradley** [comment]
Pre-release publish run to see if we're addressed the NPM limit: https://github.com/pulumi/pulumi-azure-native/actions/runs/4386814547

**Review by squaremo — CHANGES_REQUESTED** [review]
The entry for `SecurityFabric` in removed.json does not seem to have resulted in it being removed: https://github.com/pulumi/pulumi-azure-native/tree/reduce-sdk-size/sdk/dotnet/ServiceFabric/V20230201Preview

File: provider/pkg/openapi/discover.go [hunk 1]
```diff
@@ -99,6 +99,23 @@ func ReadVersions(namespace string) AzureProviders {
...
 	}
 
+	removed, err := ReadRemoved()
+	if err != nil {
+		panic(err)
+	}
+
+	for providerName, versionMap := range providers {
+		// Remove all versions that are not in the curated list.
+		if removed[providerName] != nil {
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply provider/pkg/openapi/discover.go:109]: Perhaps, what impact does it have?

**Comment by danielrbradley** [comment]
@squaremo are these really blocking changes? This seems pretty urgent to get merged so would prefer not having to go round another loop of CI for trivialities.

File: provider/pkg/openapi/versioner.go [hunk 2]
```diff
@@ -31,6 +31,16 @@ func ReadDeprecated() (ProviderVersionList, error) {
 	return readProviderVersionList(deprecatedPath)
 }
 
+func ReadRemoved() (ProviderVersionList, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	deprecatedPath := filepath.Join(dir, "versions", "removed.json")
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply provider/pkg/openapi/versioner.go:40]: Just a refactoring left-over

**Comment by squaremo** [comment]
> are these really blocking changes?

Not the change of variable name, and the map presence test -- those were me forgetting I'd made the comments when I clicked "request changes". Sorry about that.

But the removal code apparently not removing one of the specified APIs needs investigation, I think. Am I misinterpreting what it does?

**Pull request merged**